homeassistant/components/insteon/utils.py

```diff
@@ -0,0 +1,232 @@
+    EVENT_BUTTON_OFF,
+    EVENT_BUTTON_ON,
+    EVENT_CONF_BUTTON,
+    INSTEON_ENTITIES,
+    SIGNAL_LOAD_ALDB,
+    SIGNAL_PRINT_ALDB,
+    SRV_ADD_ALL_LINK,
+    SRV_ALL_LINK_GROUP,
+    SRV_ALL_LINK_MODE,
+    SRV_CONTROLLER,
+    SRV_DEL_ALL_LINK,
+    SRV_HOUSECODE,
+    SRV_LOAD_ALDB,
+    SRV_LOAD_DB_RELOAD,
+    SRV_PRINT_ALDB,
+    SRV_PRINT_IM_ALDB,
+    SRV_SCENE_OFF,
+    SRV_SCENE_ON,
+    SRV_X10_ALL_LIGHTS_OFF,
+    SRV_X10_ALL_LIGHTS_ON,
+    SRV_X10_ALL_UNITS_OFF,
+)
+from .ipdb import IPDB
+from .schemas import (
+    ADD_ALL_LINK_SCHEMA,
+    DEL_ALL_LINK_SCHEMA,
+    LOAD_ALDB_SCHEMA,
+    PRINT_ALDB_SCHEMA,
+    TRIGGER_SCENE_SCHEMA,
+    X10_HOUSECODE_SCHEMA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def register_new_device_callback(hass, config, insteon_modem):
+    """Register callback for new Insteon device."""
+
+    def _fire_button_on_off_event(address, group, val):
+        # Firing an event when a button is pressed.
+        device = insteon_modem.devices[address.hex]
+        state_name = device.states[group].name
+        button = (
+            "" if state_name == BUTTON_PRESSED_STATE_NAME else state_name[-1].lower()
+        )
+        schema = {CONF_ADDRESS: address.hex}
+        if button != "":
+            schema[EVENT_CONF_BUTTON] = button
+        if val:
+            event = EVENT_BUTTON_ON
+        else:
+            event = EVENT_BUTTON_OFF
+        _LOGGER.debug(
+            "Firing event %s with address %s and button %s", event, address.hex, button
+        )
+        hass.bus.fire(event, schema)
+
+    @callback
+    def async_new_insteon_device(device):
+        """Detect device from transport to be delegated to platform."""
+        ipdb = IPDB()
+        for state_key in device.states:
+            platform_info = ipdb[device.states[state_key]]
+            if platform_info and platform_info.platform:
+                platform = platform_info.platform
+
+                if platform == "on_off_events":
+                    device.states[state_key].register_updates(_fire_button_on_off_event)
+
+                else:
+                    _LOGGER.info(
+                        "New INSTEON device: %s (%s) %s",
+                        device.address,
+                        device.states[state_key].name,
+                        platform,
+                    )
+
+                    hass.async_create_task(
+                        discovery.async_load_platform(
+                            hass,
+                            platform,
+                            DOMAIN,
+                            discovered={
+                                "address": device.address.id,
+                                "state_key": state_key,
+                            },
+                            hass_config=config,
+                        )
+                    )
+
+    insteon_modem.devices.add_device_callback(async_new_insteon_device)
+
+
+def register_services(hass, config, insteon_modem):
```

Comment:
This could be decorated with `@callback` and renamed to `async_register_services`. There's no I/O inside as it looks.

Reply:
Are you indicating this method should become:
```
@callback
async def async_register_services(hass, config, insteon_modem):
```

or

```
@callback
def async_register_services(hass, config, insteon_modem):
```

I have tried both but the call hands even after changing the registration method to 
`hass.services.async_register`

What am I doing wrong?

Reply:
Nevermind, stupid issue..